eql/ast.py

```diff
@@ -890,30 +890,35 @@ def _render(self):
 class SubqueryBy(EqlNode):
     """Node for holding the :class:`~EventQuery` and parameters to join on."""
 
-    __slots__ = 'query', 'join_values', 'fork',
+    __slots__ = 'query', 'join_values', 'data',
 
-    def __init__(self, query, join_values=None, fork=None):
+    def __init__(self, query, join_values=None, data=None):
         """Init.
 
         :param EventQuery query: The event query enclosed in the term
         :param list[Expression] join_values: The field to join values on
-        :param bool fork: Toggle for copying instead of moving a sequence on match
+        :param bool data: Toggle for copying instead of moving a sequence on match
         """
         self.query = query
         self.join_values = join_values or []
-        self.fork = fork
+        self.data = data
 
     @property
     def params(self):
         """Keep params for backwards compatibility."""
         params = {}
-        if self.fork is not None:
-            params["fork"] = Boolean(self.fork)
+        if self.data is not None:
+            if "fork" in self.data:
+                params["fork"] = Boolean(self.data["fork"])
```

Comment:
You mean the type hint docstring? Yup, I should update that to be a little more clear.

Reply:
Yes, the docstring part for line 900, but also on line 912 where you call `self.data["fork"]` where the logic used to be `self.fork` and I was not sure where the addition of fork into the dictionary of `self.data` was done such that this would work.